### PR TITLE
Implemented: Migrated from moment timezone to luxon(#2dewerv)

### LINKED ISSUE #183: Replaced moment by luxon
## What is the motivation for adding/enhancing this feature?

As moment is in maintenance mode, replace moment by luxon.

## What are the acceptance criteria?

- [ ] ...

## Can you complete this feature request by yourself?

- [ ] YES
- [ ] NO

### PR DESCRIPTION
**Pull request by shashwatbangar**
### Related Issues

Closes #183 

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/receiving#contribution-guideline)

### CODE REVIEW

File: src/utils/index.ts [hunk 1]
```diff
@@ -30,4 +31,12 @@ const copyToClipboard = async (text: string) => {
     });
 }
 
-export { showToast, hasError, copyToClipboard, productHelpers }
+const handleDateTimeInput = (dateTimeValue: any) => {
+  // TODO Handle it in a better way
+  // Remove timezone and then convert to timestamp
+  // Current date time picker picks browser timezone and there is no supprt to change it
+  const dateTime = DateTime.fromISO(dateTimeValue, { setZone: true}).toFormat("yyyy-MM-dd'T'HH:mm:ss")
```
Comment (disha1202) [inline src/utils/index.ts:38]:
```suggestion
  const dateTime = DateTime.fromISO(dateTimeValue, { zone: 'utc' }).toFormat("yyyy-MM-dd'T'HH:mm:ss")
```

**Review by disha1202 — CHANGES_REQUESTED** [review]

File: src/utils/index.ts (hunk 1 above)
Comment (shashwatbangar) [reply src/utils/index.ts:38]: Done.

File: package.json [hunk 2]
```diff
@@ -18,10 +18,12 @@
...
     "@ionic/vue-router": "6.2.9",
     "@shopify/app-bridge-utils": "^2.0.4",
+    "@types/luxon": "^3.2.0",
     "axios": "^0.21.1",
     "axios-cache-adapter": "^2.7.3",
     "core-js": "^3.6.5",
     "http-status-codes": "^2.1.4",
+    "luxon": "^3.2.1",
     "mitt": "^2.1.0",
     "moment": "^2.29.1",
```
Comment (adityasharma7) [inline package.json:28]: I think we should remove moment related dependencies

File: src/store/modules/user/actions.ts [hunk 3]
```diff
@@ -83,7 +82,7 @@ const actions: ActionTree<UserState, RootState> = {
   async getProfile ( { commit, dispatch }) {
     const resp = await UserService.getProfile()
     if (resp.status === 200) {
-      const localTimeZone = moment.tz.guess();
+      const localTimeZone = DateTime.now().zoneName;
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:85]: We haven't set the default Luxon timezone
`Settings.defaultZone = this.userProfile.userTimeZone;`

Comment (adityasharma7) [reply src/store/modules/user/actions.ts:85]: Also, when we are setting the timezone
https://github.com/hotwax/receiving/pull/179/files#diff-d3c10a6ece79180d52468851ebe745d66365adafc59ee774a50b5687b3d42ff6R128

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/user/actions.ts [hunk 4]
```diff
@@ -83,7 +82,7 @@ const actions: ActionTree<UserState, RootState> = {
...
       if (resp.data.userTimeZone !== localTimeZone) {
         emitter.emit('timeZoneDifferent', { profileTimeZone: resp.data.userTimeZone, localTimeZone});
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:87]: We should remove feature to show timezone different alert

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

**Comment by adityasharma7** [comment]
It seems we do have support to change timezone in this app, please add support for it

**Pull request merged**